File: JanperCab/3-Application/Dtos/DuraformDesign/DuraformDesignDto.cs
using _3_Application.Dtos.DuraformEdgeProfile;
using _3_Application.Dtos.DuraformSerie;
using System.Collections.ObjectModel;

namespace _3_Application.Dtos.DuraformDesign
{
    public class DuraformDesignDto
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string ImageUrl { get; set; }

        public bool IsPopular { get; set; }

        public int DuraformSerieId { get; set; }

        public int DefaultEdgeProfileId { get; set; }

        public bool HasNoArch { get; set; }

        public decimal Thickness { get; set; }


        public DuraformSerieDto DuraformSerie { get; set; }

        public DuraformEdgeProfileDto DefaultEdgeProfile { get; set; }

        public Collection<DuraformDesignEdgeProfileDto> AllowedEdgeProfiles { set; get; }
    }
}

File: JanperCab/5-JanperCab/ClientApp/src/app/_models/duraform-design/DuraformDesignDto.ts
import { DuraformSerieDto } from './../duraform-serie/DuraformSerieDto';
import { DuraformEdgeProfileDto } from '../duraform-edge-profile/DuraformEdgeProfileDto';
import { DuraformDesignEdgeProfileDto } from './DuraformDesignEdgeProfileDto';

export class DuraformDesignDto {
  id: number;
  name: string;
  imageUrl: string;
  isPopular: boolean;
  duraformSerieId: number;
  duraformEdgeProfileId: number;
  hasNoArch: boolean;
  thickness: number;

  duraformSerie: DuraformSerieDto;
  defaultEdgeProfile: DuraformEdgeProfileDto;
  allowedEdgeProfiles: DuraformDesignEdgeProfileDto[];
}
